### PR TITLE
CodeJam-3 CoreJS

### PR DESCRIPTION
**Pull request by OlgaLazarevich**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,34 @@
+{
+  "name": "codejam",
+  "version": "1.0.0",
+  "eslintConfig": {
```
Comment (Naooki) [inline package.json]: You don't need `eslintConfig` property in _packages.json_ if you have `.eslintrc` file with the same rules.
And you have one😊

File: package.json [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  "license": "ISC",
+  "bugs": {
+    "url": "https://github.com/OlgaLazarevich/rsschool-codejam/issues"
+  },
+  "homepage": "https://github.com/OlgaLazarevich/rsschool-codejam#readme",
+  "dependencies": {},
+  "devDependencies": {
+    "eslint": "^5.8.0",
+    "eslint-config-airbnb": "^17.1.0",
+    "eslint-config-airbnb-base": "file:node_modules/eslint-config-airbnb-base",
```
Comment (Naooki) [inline package.json]: That's a bit odd...
`"eslint-config-airbnb": "^17.1.0",`
`"eslint-config-airbnb-base": "file:node_modules/eslint-config-airbnb-base",`
Instead of this you could simply install just `eslint-config-airbnb-base` with the latest version.

File: task/make.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
+module.exports = function make(a) {
+  const newArr = [a];
+  function f(b) {
+    if (typeof (b) === 'function') {
+      return newArr.reduce(b);
+    }
+    const args = [...arguments];
```
Comment (Naooki) [inline task/make.js]: ESLint highlights the error here... We should avoid directly accessing the `arguments` pseudo-array.
I assume that you probably just didn't find out how it could be done differently. So here are the tips:

**1)** the ESLint error says: `[eslint] Use the rest parameters instead of 'arguments'.`👌
You can read about the `rest parameters` here: 
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/rest_parameters

**2)** If, for some special reason, you need to violate the ESlint rule in a file, let's say, to use `arguments` pseudo-array, you can **disable** a specific rule for a file or for an entire project.
https://eslint.org/docs/user-guide/configuring#disabling-rules-with-inline-comments
https://eslint.org/docs/user-guide/configuring#disabling-rules-with-inline-comments
**Remember** that this configurations should be used only for specific cases, avoid using them.

File: test/make.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
+const assert = require('assert');
+const make = require('../task/make.js');
+
+describe('make', () => {
+  it('0', () => {
+    const count = make(15)(34, 21, 666)(41)((a, b) => a + b);
+    assert.deepEqual(count, 777);
+  });
+  it('1', () => {
```
Comment (Naooki) [inline test/make.js:9]: The point of having several test for the same code (same function) is to test **different scenarios**
The tests 1 - 4 in your case basically test the same scenario. Which means that 3 of them are actually redundant.
To make these tests more meaningful, you could test some diffeferent scenarios, let's say:
- `make(15, 34, 21, 666, 41, (a, b) => a + b);`  // all parameters in a one call
- `make((a, b) => a + b);`  // only the function
- `make();`  // the **negative** test. The function should work without errors.
- `make(15)(1)((a, b) => a * b);`  // works as expected with some different function.

Basically, the more **different** tests you have, the more chances you will detect the incorrect behavior in your code. And therefore, less chances that it would be found by somebody else 😎
